tests/lib/devicetree/api/src/main.c

```diff
@@ -2229,6 +2236,17 @@ static void test_string_token(void)
 		zassert_true(array_inst[i] != token_no_inst, "");
 	}
 
+	/* Test DT_INST_STRING_UPPER_TOKEN_OR when property is found */
```

Comment:
It looks like this is missing a test for DT_INST_STRING_TOKEN and DT_INST_STRING_UPPER_TOKEN. We should have tests for every macro in devicetree.h, even if right now they're implemented in terms of something else. This is so that we avoid regressions in case that implementation detail changes in the future.